File: day1/day1.py
# DAY 1: Fuel required to launch a given module is based on its mass. 
# Specifically, to find the fuel required for a module, take its mass, 
# divide by three, round down, and subtract 2.

f = open("input.txt", "r")

# part one
total = 0

for mass in f:
    fuel = (int(mass) // 3) - 2 # // for integer division
    total += fuel

f.close()
print("part one fuel needed: ", total)

# part two
f = open("input.txt", "r")

total = 0
for mass in f:
    fuel = (int(mass) // 3) - 2
    while fuel > 0:
        total += fuel
        fuel = (fuel // 3) - 2
        
    
f.close()
print("part two fuel needed: ", total)

File: day1/day1.cpp
// DAY 1: Fuel required to launch a given module is based on its mass. 
// Specifically, to find the fuel required for a module, take its mass, 
// divide by three, round down, and subtract 2.

#include <iostream>  
#include <fstream>
  
using namespace std;

void part_one(ifstream&);
void part_two(ifstream&);

int main() { 
    ifstream input ("input.txt");
    if (input.is_open()) {
        //part_one(input);
        part_two(input);
    }
    input.close();
    return 0;
} 

void part_one(ifstream& input) {
    int sum = 0;
    int line;

    while (input >> line) {
        sum += (line / 3) - 2;
    }

    cout << sum << "\n";
}

void part_two(ifstream& input) {
    int sum = 0;
    int line;

    while (input >> line) {
        line = (line / 3) - 2;
        while (line > 0) {
            sum += line;
            line = (line / 3) - 2;
        }
    }

    cout << sum << "\n";
}
